nifi-api/src/main/java/org/apache/nifi/components/PropertyDescriptor.java

```diff
@@ -313,6 +313,27 @@ public Builder defaultValue(final String value) {
             return this;
         }
 
+        /**
+         * Specifies the initial value and the default value that will be used
+         * if the user does not specify a value. When {@link #build()} is
+         * called, if Allowable Values have been set (see
+         * {@link #allowableValues(AllowableValue...)})
+         * and the "Value" of the {@link DescribedValue} object is not
+         * the "Value" of one of those Allowable Values, an Exception will be thrown.
+         * If the Allowable Values have been set using the
+         * {@link #allowableValues(AllowableValue...)} method, the default value
+         * should be set providing the {@link AllowableValue} to this method.
+         *
+         * @param describedValue default value holder
+         * @return the builder
+         */
+        public Builder defaultDescribedValue(final DescribedValue describedValue) {
```

Comment:
Yes, this was my initial idea as well. However, this prohibits the user of `.defaultValue(null)` as the compiler cannot differentiate in that case. 
If we're okay with that, I'm more than happy to make it an overloaded method. What do you think?

Reply:
I definitely think it's fine to require `.defaultValue((String) null)`. But I'm not sure this is ever needed, since `null` is always the default unless otherwise specified.